### PR TITLE
shot chance

### PR DESCRIPTION
**Pull request by houghta4**
WIP formula for enemy shot chance. Will need to see if this fits with the speed of everything else. Right now the only factor in shooting is being on the screen and how far the enemy is from the player

### CODE REVIEW

**Comment by houghta4** [comment]
I think for just a firing chance that this is good to base it off of distance. Possibly in the future we can have different enemies with different firing rates/ accuracy. eg sniper vs grunt. What do you think of the speed of this?

**Comment by nnickodem** [comment]
I think its good enough for now, may need to adjust in while building the level but is a pretty simple adjustment

File: src/game/enemy/systems.rs [hunk 1]
```diff
@@ -214,49 +209,54 @@ pub fn catch_shot_event(
...
+            //#endregion
         }
     }
 }
 
+/// Scale shot chance with distance from enemy to player
+/// * `dist` is the Euclidean distance between the enemy and the player
+/// ### Return true if enemy should fire else false
+fn shot_chance(dist: f32) -> bool {
+    let chance = 1.0 / (10.0 * dist); // bigger distance -> lower chance
```
Comment (nnickodem) [inline src/game/enemy/systems.rs:256]: we will need to make this happen more often, hardly happens at all with this

**Review by nnickodem — CHANGES_REQUESTED** [review]

**Comment by houghta4** [comment]
I doubled the chance. I wasn't sure how fast they should be firing since we need to hit an enemy 5 times (plus misses) to kill it. This feels a little better for me with your changes as well. We can keep tweaking if needed

**Review by nnickodem — APPROVED** [review]

**Pull request merged**